src/main/java/com/google/devtools/build/lib/skylarkbuildapi/repository/SkylarkRepositoryContextApi.java

```diff
@@ -294,6 +294,31 @@ public SkylarkExecutionResultApi execute(
   public boolean delete(Object path, Location location)
       throws EvalException, RepositoryFunctionExceptionT, InterruptedException;
 
+  @SkylarkCallable(
+      name = "patch",
+      doc =
+          "Apply a patch file to the external repository.",
```

Comment:
I think we should mention something like the root directory of external repository... Also, do we support only some specific patch format? We should mention it in this doc string.